addon/pods/services/notifier-service.js

```diff
@@ -0,0 +1,99 @@
+import Ember from 'ember'
+
+/**
+ * @typedef {Object} NotificationOption
+ * @property {String} message - message to display
+ * @property {Object|String} details - details to pass to onDetailsClick
+ * @property {Function} onDetailsClick - callback for when the details is clicked
+ * @property {String} type - type of notification (success, info, warning, error)
+ * @property {Boolean} autoClear - whether to dismiss notification automatically
+ * @property {Number} clearDuration - time in ms to dismiss if autoCleared
+ */
+
+/**
+ * Manages all notifications
+ */
+export default Ember.Service.extend({
+  init: function () {
+    this.notifications = Ember.A()
+  },
+
+  /**
+   * default value to use for clearDuration
+   * @type {Number}
+   */
+  defaultClearDuration: 5000,
+
+  /**
+   * Adds a notifications
+   * @param {NotificationOption} options - notification options
+   * @returns {Object} the Emberified notification object
+   */
+  addNotification: function (options) {
+    // If no message is set, throw an error
+    if (!options.message) {
+      throw new Error('No notification message set')
+    }
+
+    var notification = Ember.Object.create({
```

Comment:
Are you a discriminating against `var`. smh ;)

Don't worry, I'll try and find them all and eliminate them.